pkg/controller/nodes/transformers.go

```diff
@@ -79,7 +87,10 @@ func ToNodeExecutionEvent(nodeExecID *core.NodeExecutionIdentifier,
 	status v1alpha1.ExecutableNodeStatus,
 	eventVersion v1alpha1.EventVersion,
 	parentInfo executors.ImmutableParentInfo,
-	node v1alpha1.ExecutableNode, clusterID string, dynamicNodePhase v1alpha1.DynamicNodePhase) (*event.NodeExecutionEvent, error) {
+	node v1alpha1.ExecutableNode,
+	clusterID string,
+	dynamicNodePhase v1alpha1.DynamicNodePhase,
+	dataStore *storage.DataStore) (*event.NodeExecutionEvent, error) {
```

Comment:
I don't think we should pass `dataStore` here...

Can you, instead, add DeckURI to [OutputInfo](https://github.com/flyteorg/flytepropeller/blob/59069d2aee79de353b2e1cc24c1912c89335faab/pkg/controller/nodes/handler/transition_info.go#L52). 
Then populate it wherever we populate OutputURI field?